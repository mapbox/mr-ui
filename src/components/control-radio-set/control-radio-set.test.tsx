import React from 'react';
import ControlRadioSet from './control-radio-set';
import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ControlRadioSet', () => {
  describe('basic', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      options: [
        { label: 'Option 1', value: 'Option value 1' },
        { label: 'Option 2', value: 'Option value 2' }
      ],
      onChange: mockOnChange
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRadioSet {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange gets called with input id and value', async () => {
      render(<ControlRadioSet {...props} />)
      await userEvent.click(screen.getByLabelText('Option 1'));

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(1);
      });

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledWith('Option value 1', 'testinput');
      });
    });
  });

  describe('disabled', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      options: [
        { label: 'Option 1', value: 'Option value 1', disabled: true },
        { label: 'Option 2', value: 'Option value 2' }
      ],
      onChange: mockOnChange
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRadioSet {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange is *not* called with input id and value', async () => {
      render(<ControlRadioSet {...props} />)

      await userEvent.click(screen.getByTestId('testinput-0-input'));
      expect(screen.getByTestId('testinput-0-input')).toBeDisabled();

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(0);
      });
    });
  });

  describe('all options', () => {
    const props = {
      id: 'testinput',
      options: [
        { label: <span>Option 1</span>, value: 'Option value 1' },
        { label: <span>Option 2</span>, value: 'Option value 2' },
        { label: <span>Option 3</span>, value: 'Option value 3' }
      ],
      validationError: 'oh no!',
      value: 'Option value 1',
      autoFocus: true,
      optional: true,
      themeRadioContainer: 'w-full',
      themeRadio: 'radio--white radio--active-red mr6',
      themeControlWrapper: 'bg-red-light',
      onChange: jest.fn()
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRadioSet {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });
});
