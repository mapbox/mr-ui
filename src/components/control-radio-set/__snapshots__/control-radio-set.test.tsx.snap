// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlRadioSet all options renders 1`] = `
<body>
  <div>
    <div
      class="relative bg-red-light"
    >
      <div
        aria-invalid="true"
        aria-required="false"
        data-testid="testinput-input"
        id="testinput"
        role="radiogroup"
      >
        <label
          class="radio-container w-full"
        >
          <input
            checked=""
            data-testid="testinput-0-input"
            name="testinput"
            type="radio"
            value="Option value 1"
          />
          <div
            class="radio--white radio--active-red mr6 radio"
          />
          <span>
            Option 1
          </span>
        </label>
        <label
          class="radio-container w-full"
        >
          <input
            data-testid="testinput-1-input"
            name="testinput"
            type="radio"
            value="Option value 2"
          />
          <div
            class="radio--white radio--active-red mr6 radio"
          />
          <span>
            Option 2
          </span>
        </label>
        <label
          class="radio-container w-full"
        >
          <input
            data-testid="testinput-2-input"
            name="testinput"
            type="radio"
            value="Option value 3"
          />
          <div
            class="radio--white radio--active-red mr6 radio"
          />
          <span>
            Option 3
          </span>
        </label>
      </div>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="testinput-error"
          id="testinput-error"
        >
          oh no!
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlRadioSet basic renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        aria-required="true"
        data-testid="testinput-input"
        id="testinput"
        role="radiogroup"
      >
        <label
          class="radio-container txt-s block mb6 flex"
        >
          <input
            data-testid="testinput-0-input"
            name="testinput"
            type="radio"
            value="Option value 1"
          />
          <div
            class="mr6 radio--blue radio--s-label inline-block radio"
          />
          Option 1
        </label>
        <label
          class="radio-container txt-s block mb6 flex"
        >
          <input
            data-testid="testinput-1-input"
            name="testinput"
            type="radio"
            value="Option value 2"
          />
          <div
            class="mr6 radio--blue radio--s-label inline-block radio"
          />
          Option 2
        </label>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;

exports[`ControlRadioSet disabled renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        aria-required="true"
        data-testid="testinput-input"
        id="testinput"
        role="radiogroup"
      >
        <label
          class="radio-container txt-s block mb6 flex"
        >
          <input
            data-testid="testinput-0-input"
            disabled=""
            name="testinput"
            type="radio"
            value="Option value 1"
          />
          <div
            class="mr6 radio--blue radio--s-label inline-block radio"
          />
          Option 1
        </label>
        <label
          class="radio-container txt-s block mb6 flex"
        >
          <input
            data-testid="testinput-1-input"
            name="testinput"
            type="radio"
            value="Option value 2"
          />
          <div
            class="mr6 radio--blue radio--s-label inline-block radio"
          />
          Option 2
        </label>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;
